test/snapshot.spec.ts

```diff
@@ -0,0 +1,95 @@
+import { H256 } from "codechain-primitives/lib";
+import { Block, MintAsset } from "codechain-sdk/lib/core/classes";
+import models from "../src/models";
+import * as BlockModel from "../src/models/logic/block";
+import * as UTXOModel from "../src/models/logic/utxo";
+import * as Helper from "./helper";
+
+let mintBlock: Block;
+let transferBlock: Block;
+let assetType: H256;
+
+beforeAll(async done => {
+    // Sync the genesis block.
+    await Helper.worker.sync();
+    await Helper.runExample("import-test-account");
+    await Helper.runExample("mint-and-transfer");
+    const bestBlockNumber = await Helper.sdk.rpc.chain.getBestBlockNumber();
+    const mintBlockNumber = bestBlockNumber - 1;
+    mintBlock = (await Helper.sdk.rpc.chain.getBlock(
+        mintBlockNumber
+    ))!;
+    const assetMintTransaction = mintBlock.transactions[0].unsigned as MintAsset;
+    assetType = assetMintTransaction.getAssetSchemeAddress();
+    transferBlock = (await Helper.sdk.rpc.chain.getBlock(
+        bestBlockNumber
+    ))!;
+    done();
+}, 30_000);
+
+afterAll(async done => {
+    await models.sequelize.close();
+    done();
+});
+
+test("Get an empty snapshot", async done => {
+    const snapshot = await UTXOModel.getSnapshot(assetType, mintBlock.number);
+    expect(snapshot).toHaveLength(0);
+    done();
+});
+
+test("Sync", async done => {
+    const prevBlockCount = await BlockModel.getNumberOfBlocks({});
+    await Helper.worker.sync();
+    const blockCount = await BlockModel.getNumberOfBlocks({});
+    expect(blockCount).toBe(prevBlockCount + 2);
+    done();
+});
+
+test("Get the mint snapshot", async done => {
+    const snapshot = await UTXOModel.getSnapshot(assetType, mintBlock.number);
+    expect(snapshot).toHaveLength(1);
+    expect(snapshot[0].blockNumber).toBe(mintBlock.number);
+    expect(snapshot[0].transactionHash).toBe(mintBlock.transactions[0].hash().value);
+    expect(snapshot[0].usedBlockNumber).toBeNull();
+    expect(snapshot[0].usedTransactionHash).toBeNull();
+    done();
+});
+
+test("Fail to get the transfer block by timestamp", async done => {
+    const block = await BlockModel.getByTime(transferBlock.timestamp);
+    expect(block).toBeNull();
+    done();
+})
+
+test("Add another block", async done => {
+    const prevBlockCount = await BlockModel.getNumberOfBlocks({});
+    await Helper.runExample("send-signed-tx");
+    await Helper.worker.sync();
+    const blockCount = await BlockModel.getNumberOfBlocks({});
+    expect(blockCount).toBe(prevBlockCount + 1);
+    done();
+})
+
+test("Get the transfer block by timestamp", async done => {
+    const block = await BlockModel.getByTime(transferBlock.timestamp);
+    expect(block).toBeTruthy();
+    done();
+})
+
+test("Get the transfer snapshot", async done => {
+    const snapshot = await UTXOModel.getSnapshot(assetType, transferBlock.number);
+    expect(snapshot).toHaveLength(2);
+
+    for (let blockNumber = 0; blockNumber < 2; blockNumber++) {
```

Comment:
Why do you check it twice?

Reply:
I made a mistake.
I was going to check two UTXOs in the snapshot. I'll fix it now.

Reply:
I fixed it.